### PR TITLE
[spaceship] Add support for providing group names for testers and creating new groups

### PR DESCRIPTION
**Pull request by KrauseFx**
Use the following code to use an existing group based on the ID and to create or use a group based on the name

```ruby
Spaceship::Tunes::Tester.external.create!(
  email: "something@krausefx.com",  
  groups: ["b6f65dbd-c845-4d91-bc39-0b661d608970"], 
  group_names: ["CustomGroup"])
```

### CODE REVIEW

**Comment by asfalcone** [comment]
Are there any specs or fixture data to update or create for this change?

**Comment by KrauseFx** [comment]
I don't think we have any tests for adding testers currently, I'll add some.

To review this PR, use the `?w=1`: https://github.com/fastlane/fastlane/pull/7788/files?w=1

**Comment by KrauseFx** [comment]
@asfalcone Updated the PR with tests 👍

**Comment by asfalcone** [comment]
Does this test the group names functionality that you added?

File: spaceship/lib/spaceship/tunes/tunes_client.rb [hunk 1]
```diff
@@ -831,26 +831,46 @@ def groups
...
+        tester_data[:groups] += groups.map do |x|
+          {
+            "id" => x,
+            "isSelected" => true
+          }
+        end
+      end
+
+      if group_names
+        tester_data[:groups] += group_names.map do |x|
```
Comment (hjanuschka) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: can we get rid of `x` ?

File: spaceship/lib/spaceship/tunes/tunes_client.rb [hunk 2]
```diff
@@ -831,26 +831,46 @@ def groups
...
+          value: last_name || ""
+        },
+        testing: {
+          value: true
+        },
+        groups: []
+      }
+
+      if groups
+        tester_data[:groups] += groups.map do |x|
```
Comment (hjanuschka) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: can we get rid of `x` ?

File: spaceship/lib/spaceship/tunes/tunes_client.rb [hunk 3]
```diff
@@ -831,26 +831,46 @@ def groups
...
+            "isSelected" => true
+          }
+        end
+      end
+
+      if group_names
+        tester_data[:groups] += group_names.map do |x|
+          {
+            "name" => {
+              "value" => x
```
Comment (hjanuschka) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: does that work with just passing in names instead of id's ?

**Review by hjanuschka — COMMENTED** [review]
looking good to me - left a couple of comments.

File: spaceship/lib/spaceship/tunes/tunes_client.rb (hunk 2 above)
Comment (KrauseFx) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: Haha, yeah, I have no idea why I used x here

File: spaceship/lib/spaceship/tunes/tunes_client.rb (hunk 3 above)
Comment (KrauseFx) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: No you have to provide either or

**Comment by KrauseFx** [comment]
Whops, I don't know why I kept this open, this was implemented in https://github.com/fastlane/fastlane/pull/8005

**Pull request closed without merging**